File: rules-interpreter/src/main/java/io/github/iamrenny/ruleflow/evaluators/ValueContextEvaluator.java
package io.github.iamrenny.ruleflow.evaluators;

import io.github.iamrenny.ruleflow.RuleFlowLanguageParser.ValueContext;
import io.github.iamrenny.ruleflow.visitors.Visitor;
import io.github.iamrenny.ruleflow.utils.DateTimeUtils;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import java.time.LocalDateTime;


public class ValueContextEvaluator implements ContextEvaluator<ValueContext> {
    private static final Logger logger = LoggerFactory.getLogger(ValueContextEvaluator.class);

    @Override
    public Object evaluate(ValueContext ctx, Visitor visitor) {
        Object result;
        if (ctx.validValue().string != null) {
            result = (ctx.validValue().string.getText()).replace("'", "");
        } else if (ctx.validValue().number != null) {
            result = Double.valueOf(ctx.validValue().number.getText());
        } else if (ctx.validValue().nullValue != null) {
            result = ctx.validValue().nullValue;
        } else if (ctx.validValue().currentDate != null) {
            result = LocalDateTime.now();
        } else if (ctx.validValue().booleanLiteral != null) {
            result = Boolean.parseBoolean(ctx.validValue().booleanLiteral.getText());
        } else {
            throw new IllegalArgumentException("Value not supported: " + ctx.getText());
        }
        logger.debug("Value: ctx={}, result={}", ctx.getText(), result);
        return result;
    }
}

File: rules-interpreter/src/main/kotlin/com/rappi/fraud/rules/parser/evaluators/ValueContextEvaluator.kt
package com.rappi.fraud.rules.parser.evaluators

import com.rappi.fraud.analang.ANAParser.ValueContext
import com.rappi.fraud.rules.parser.visitors.Visitor
import com.rappi.fraud.rules.parser.removeSingleQuote
import java.time.LocalDateTime

class ValueContextEvaluator : ContextEvaluator<ValueContext> {

    override fun evaluate(ctx: ValueContext, visitor: Visitor): Any {
        return when {
            ctx.validValue().string != null -> ctx.validValue().string.text.removeSingleQuote()
            ctx.validValue().number != null -> ctx.validValue().number.text.toBigDecimal()
            ctx.validValue().nullValue != null -> ctx.validValue().nullValue
            ctx.validValue().currentDate != null -> LocalDateTime.now()
            else -> throw IllegalArgumentException("Value not supported: ${ctx.text}")
        }
    }
}